components/FileMenu.vue

```diff
@@ -143,71 +142,27 @@ const closeModal = () => {
 
 watch(errorMessages, openModal);
 
-const handleMenuPosition = () => {
-  if (wasFileDisplayed.value) {
-    const windowInnerHeight = window.innerHeight;
-    const rectBottom = root.value?.getBoundingClientRect().bottom || 0;
-    const bottomDistance = windowInnerHeight - rectBottom;
-    if (bottomDistance < 200) {
-      menuListPosition.value = "top";
-    } else {
-      menuListPosition.value = "bottom";
-    }
-  }
-};
-
-const throttle = (fn: Function, delay: number) => {
-  let time = Date.now();
-  return () => {
-    if (time + delay - Date.now() <= 0) {
-      fn();
-      time = Date.now();
-    }
-  };
-};
-
 onMounted(() => {
-  const fileListElement = document.querySelector(".files__list");
   const target = root.value as Element;
 
-  const createObserver = () => {
-    const observerOptions = {
-      root: fileListElement,
-      rootMargin: "0px",
-      threshold: [0, 0.5, 1],
-    };
-
-    const observer = new IntersectionObserver(
-      intersectionCallback,
-      observerOptions
-    );
-
-    observer.observe(target);
+  const observerOptions = {
+    rootMargin: "0px 0px -240px 0px",
+    threshold: [0, 0.25, 0.5, 0.75, 1],
   };
 
-  const intersectionCallback = (observer: IntersectionObserverEntry[]) => {
-    observer.forEach((target) => {
-      if (target.isIntersecting) {
-        wasFileDisplayed.value = true;
-        handleMenuPosition();
+  const observer = new IntersectionObserver((entries) => {
+    entries.forEach((entry) => {
+      if (entry.intersectionRatio < 1 && entry.boundingClientRect.top > 240) {
+        menuListPosition.value = "top";
+      } else if (!entry.isIntersecting && entry.boundingClientRect.top > 900) {
+        observer.unobserve(target);
```

Comment:
if you unobserve here - will it work next time if it will be intersected? 🤔 seems like it work until it is unobserved for the first time (or maybe I'm wrong 😄) - maybe safer would be to unobserve `beforeUnmount`?

Reply:
of course you are write :)

Reply:
To have access to observer in the onBeforeUnmout hook I had to create the observer ref and assign to it the new Intersection Observer. 
There is target defined twice - in onMounted and onBeforeUnmout hooks. I tried to define it outside as a ref, but even if there was no errors with types in the vsc and the console - it didn't work properly. So I decided to leave it as it works properly.